packages/rxfire/test/database.test.ts

```diff
@@ -0,0 +1,634 @@
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { initializeApp, database, app } from 'firebase';
+import { fromRef } from '../database/fromRef';
+import { list, ChildEvent } from '../database';
+import { take, skip, switchMap } from 'rxjs/operators';
+import { BehaviorSubject } from 'rxjs';
+import { auditTrail } from '../database/list/audit-trail';
+
+const rando = () =>
+  Math.random()
+    .toString(36)
+    .substring(5);
+
+let batch = (items: any[]) => {
+  let batch = {};
+  Object.keys(items).forEach(function(key) {
+    const itemValue = items[key];
+    batch[itemValue.key] = itemValue;
+  });
+  // make batch immutable to preserve integrity
+  return Object.freeze(batch);
+};
+
+describe('RxFire Database', () => {
+  let app: app.App = null;
+  let database: database.Database = null;
+  let ref = (path: string) => {
+    app.database().goOffline();
+    return app.database().ref(path);
+  };
+
+  function prepareList(
+    opts: { events?: ChildEvent[]; skipnumber: number } = { skipnumber: 0 }
+  ) {
+    const { events, skipnumber } = opts;
+    const aref = ref(rando());
+    const snapChanges = list(aref, events);
+    return {
+      snapChanges: snapChanges.pipe(skip(skipnumber)),
+      ref: aref
+    };
+  }
+
+  /**
+   * Each test runs inside it's own app instance and the app
+   * is deleted after the test runs.
+   *
+   * Database tests run "offline" to reduce "flakeyness".
+   *
+   * Each test is responsible for seeding and removing data. Helper
+   * functions are useful if the process becomes brittle or tedious.
+   * Note that removing is less necessary since the tests are run
+   * offline.
+   *
+   * Note: Database tests do not run exactly the same offline as
+   * they do online. Querying can act differently, tests must
+   * account for this.
+   */
+  beforeEach(() => {
+    app = initializeApp({
```

Comment:
Test using the user specified test config please.

Reply:
Not sure what you mean seeing how this is an automated test.

Reply:
The config here points to a static firebase project. As part of the test setup for the repo we have users provide a config. I want us to reference that config here if we are actually going to hit prod for our tests

Reply:
fixed